docs/development-guides/creating-custom-blocks.md

```diff
@@ -0,0 +1,365 @@
+        tagName="h2"
+        placeholder={__('Custom Title')}
+        value={title}
+        onChange={(title) => setAttributes({ title })}
+      />
+    </div>
+  );
+};
+export default MyBlockEdit;
+```
+
+## Server-Side Rendering
+
+The WP Scaffold uses server-side rendering for blocks. The save.js file returns null, and the actual rendering is done in the markup.php file:
+
+```javascript
+// save.js
+/**
+ * Dynamic blocks do not save the HTML.
+ *
+ * @returns {null} Dynamic blocks do not save the HTML.
+ */
+const MyBlockSave = () => null;
+
+export default MyBlockSave;
+```
+
+```php
+// markup.php
+<?php
+/**
+ * Block markup
+ *
+ * @package TenUpTheme\Blocks\MyBlock
+ *
+ * @var array    $attributes         Block attributes.
+ * @var string   $content            Block content.
+ * @var WP_Block $block              Block instance.
+ * @var array    $context            Block context.
+ */
+
+?>
+<div <?php echo get_block_wrapper_attributes(); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
+  <h2 class="wp-block-tenup-my-block__title">
+    <?php echo wp_kses_post( $attributes['title'] ); ?>
+  </h2>
+</div>
+```
+
+## Block Styles
+
+You can add styles for your block by creating CSS files that will be automatically loaded. The build process handles this for you.
+
+For block-specific styles, they will be automatically enqueued if placed in the appropriate location:
+
+```
+dist/blocks/autoenqueue/tenup/my-block.css
+```
+
+## Block Patterns
+
+You can register block patterns to provide pre-designed layouts using your blocks:
+
+```php
+// This is handled in the Blocks.php file
+register_block_pattern_category(
+  '10up-theme',
+  [ 'label' => __( '10up Theme', 'tenup-theme' ) ]
+);
+
+// You can add your own patterns
+register_block_pattern(
+  'tenup-theme/my-pattern',
+  array(
+    'title'       => __('My Pattern', 'tenup-theme'),
+    'description' => __('A custom pattern', 'tenup-theme'),
+    'categories'  => array('10up-theme'),
+    'content'     => '<!-- wp:tenup/my-block {"title":"Pattern Title"} /-->'
+  )
+);
+```
+
+## Testing Your Block
+
+To test your block:
+
+1. Build your block assets:
+   ```bash
+   npm run build
+   ```
+
+2. Or use the watch mode during development:
+   ```bash
+   npm run watch
+   ```
+
+3. Open the WordPress editor
+4. Add your block to a post or page
+5. Test all functionality and interactions
+6. Preview the post to see how the block renders on the frontend
+
+## Advanced Techniques
+
+### InnerBlocks
+
+Allow nested blocks within your block:
+
+```javascript
+// edit.js
+import { useBlockProps, InnerBlocks } from '@wordpress/block-editor';
+
+export default function Edit() {
+  const blockProps = useBlockProps();
+
+  return (
+    <div {...blockProps}>
+      <InnerBlocks
+        allowedBlocks={['core/paragraph', 'core/image']}
+        template={[
+          ['core/paragraph', { placeholder: 'Add content...' }]
+        ]}
+      />
+    </div>
+  );
+}
+```
```

Comment:
Should this include `useInnerBlocksProps` as per our [Block Editor best practices](https://gutenberg.10up.com/reference/Blocks/inner-blocks/)?

Reply:
Yes just updated this  👍